### PR TITLE
add: spring keyvault sample starter add authority-host param

### PR DESCRIPTION
**Pull request by backwind1233**
- verify issue #12227 
- spring keyvalut starter support authority-host params

### CODE REVIEW

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java [hunk 1]
```diff
@@ -133,6 +136,8 @@ public TokenCredential getCredentials(String normalizedName) {
         final String tenantId = getPropertyValue(normalizedName, Property.TENANT_ID);
         final String certificatePath = getPropertyValue(normalizedName, Property.CERTIFICATE_PATH);
         final String certificatePassword = getPropertyValue(normalizedName, Property.CERTIFICATE_PASSWORD);
+        final String authorityHost = getPropertyValue(normalizedName, Property.AUTHORITY_HOST) == null ?
+            new IdentityClientOptions().getAuthorityHost() : getPropertyValue(normalizedName, Property.AUTHORITY_HOST);
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:140]: Is this the default value for authority host?
`new IdentityClientOptions().getAuthorityHost()`

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java [hunk 2]
```diff
@@ -79,6 +81,7 @@ public void addKeyVaultPropertySource(String normalizedName) {
         final SecretClient secretClient = new SecretClientBuilder()
                 .vaultUrl(vaultUri)
                 .credential(tokenCredential)
+                .serviceVersion(SecretServiceVersion.V7_0)
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:84]: Why do we need this value here?

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java (hunk 1 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:140]: yes,the constructor of IdentityClientOptions will initialize authorityHost  with default value

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java (hunk 2 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:84]: by default, serviceVersion value is V7_1, but the starter run failed with errors below
```
{"error":{"code":"BadParameter","message":"The specified version (7.1) is not recognized. Consider using the latest supported version (2016-10-01)."}}
```

File: sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java [hunk 3]
```diff
@@ -18,13 +18,12 @@
...
 import com.microsoft.azure.keyvault.spring.KeyVaultProperties.Property;
 import com.microsoft.azure.telemetry.TelemetrySender;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+
+import java.util.*;
+
```
Comment (zhoufenqin) [inline sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java]: we need to specify the package

**Review by saragluna — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
Discussed with @backwind1233 that seems like the service version is now using a different schema, check https://github.com/Azure/azure-rest-api-specs/tree/master/specification/keyvault/data-plane/Microsoft.KeyVault/stable. So we won't try to convert the 7.0 version to something like `2019-07-01`. The README for the authority host and service version is not complete and we'll add them in a separate PR.

**Pull request merged**